### PR TITLE
[IMP][9.0] medical_pathology: Improved data handling

### PR DESCRIPTION
**Pull request by lasley**
This change provides an isolation of pathologies and categories by their code type. This is required for usability once data for multiple code types is filled in (ICD-10, ICD-9, SNOMED, etc.)

* Add pathology code types to pathology categories
* Add domains restricting pathologies and categories to that within the type
* Update views to include default contexts and filters for code type
* Add notes field to category + create an example demo record case from ICD-10
Verified
* Add a parent child relationship on medical pathology
* Bump version

Depends:
- [x] #138

### CODE REVIEW

**Comment by lasley** [comment]
This is ready for review. Runbot error is superficial caused by OCA/runbot-addons#111

File: medical_pathology/demo/medical_pathology_demo.xml [hunk 1]
```diff
@@ -22,4 +22,44 @@
...
+    <record id="medical_pathology_medical_pathology_A00" model="medical.pathology">
+        <field name="name">Cholera</field>
+        <field name="code">A00</field>
+        <field name="code_type_id" ref="medical_pathology.pathology_code_05" />
+        <field name="category_id" ref="medical_pathology.medical_pathology_category_7" />
+    </record>
+
+    <record id="medical_pathology_medical_pathology_A00_0" model="medical.pathology">
+        <field name="name">Cholera due to Vibrio cholerae 01, biovar cholerae</field>
+        <field name="code">A00.0</field>
```
Comment (lasley) [inline medical_pathology/demo/medical_pathology_demo.xml]: These ICD10 codes will conflict when data is imported in #134. Add an underscore or something to create a unique namespace while providing somewhat relevant data for demo

**Review by lasley — COMMENTED** [review]
Note to self: Add hyphenation for the ICD code types in data (ICD10CM should be ICD-10-CM)

**Review by susport — COMMENTED** [review]
In test runbut. I create a new pathology and choose the Parent Pathology throw the error "Error: NameError: name 'default_child_ids' is not defined". In view field parent_id in context

**Comment by lasley** [comment]
Nice catch, thanks @susport - typo fixed & everything is now working smoothly on Runbot!

**Comment by rafaelbn** [comment]
Do I must expect to have a README.md?

**Review by susport — COMMENTED** [review]
Good @lasley 👍

**Comment by lasley** [comment]
@rafaelbn - I'm not sure what you mean? There shouldn't be a .md in the module, but I'm not really making any changes that would need to go in the ReadMe I think. This PR just adds a bit more datapoints to pathologies.

Thanks @susport!

**Comment by rafaelbn** [comment]
I would like to know what this module do and test It. I expected It in Readme as OCA standards. Maybe i'm tired enough to Miss something

**Comment by lasley** [comment]
@rafaelbn Ahh good call, that is 100% something everything in Vertical Medical is missing - explicit directions in the ReadMe regarding the location of things that it adds.

Honestly this is something that's never existed, and the ReadMe's we have now are totally not up to standard in terms of usage. I'll add to this PR for Pathologies, begin to get some PRs going for the modules that are already on 9, and update the ones currently in PR.

**Comment by rafaelbn** [comment]
OK! Thanks for clarify. I don't have a business need but if you have good READMES and I can understand what I must test, I will test everything 👍 .

**Comment by lasley** [comment]
Seriously I am in dire need of testers, so to take this a step further I have created #147 and will be pushing a doc fix for this module within the next hour or so.

I'm not sure why I didn't think of this earlier. I'm too thick in the murk I think.

**Comment by lasley** [comment]
@rafaelbn - Readme updated; let me know if it's still not clear. Going forward, please continue to kick the boot to our ReadMes if they aren't clear, we very much need outside perspective on that I think.

**Comment by lasley** [comment]
@t3ddftw please review

**Review by tedsalmon — APPROVED** [review]
👍  LGTM!

**Pull request merged**